From bd5f2c35a72227782db6c725baa34d492101f4e5 Mon Sep 17 00:00:00 2001
From: Alan Screen <awscreen@chromium.org>
Date: Tue, 11 Jun 2024 13:50:02 +0000
Subject: [PATCH] Update DevTools to include explicit dependency on printing mojom

DevTools source has a direct include of //chrome/browser/printing some
C++ headers, but there is no build target for it to directly reference.
This means that necessary mojom files that need to be generated for the
C++ headers might not have been generated by the time these DevTools
source files are compiled.

The //chrome/browser build is a massive target which is heavily
entangled and not easily broken into smaller, more manageable targets.
So resolve this dependency issue by including a direct dependency on
the needed //chrome/services/printing/public/mojom target.

Fixed: 345662469
Change-Id: I713ed5ae7395f55c10cb0dcf5252923805981b46
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5618290
Reviewed-by: Fumitoshi Ukai <ukai@google.com>
Reviewed-by: Alex Rudenko <alexrudenko@chromium.org>
Commit-Queue: Alan Screen <awscreen@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1313377}
---

diff --git a/chrome/browser/devtools/BUILD.gn b/chrome/browser/devtools/BUILD.gn
index 1e97efb..6dfa089 100644
--- a/chrome/browser/devtools/BUILD.gn
+++ b/chrome/browser/devtools/BUILD.gn
@@ -318,6 +318,14 @@
         "//components/printing/browser/headless",
         "//components/printing/browser/print_to_pdf",
       ]
+      if (enable_oop_printing) {
+        # protocol/page_handler.cc directly includes
+        # chrome/browser/printing/print_view_manager.h, and thus is dependent
+        # upon some mojom files.  This could go away and have a cleaner dep on
+        # //chrome/browser/printing if the //chrome build layer could get
+        # sufficiently untangled.
+        deps += [ "//chrome/services/printing/public/mojom" ]
+      }
     }
     sources += rebase_path(_protocol_generated, ".", target_gen_dir)
   }
